File: Projects/01-Name_And_Age/C++/main.cpp

#include <iostream>

int main(){
    std::string name;
    int age;
    std::cout << "Please enter your name: ";
    std::cin >> name;
    std::cout << "Please enter your age: ";
    std::cin >> age;

    std::cout << name << " is " << age << " years old "<< std::endl;

    std::string name2 = "Peter Parker";
    int age2 = 17;

    std::cout << name2 << " is " << age2 << " years old "<< std::endl;

    if(age > age2){
        std::cout << name << " is older than " << name2 << std::endl;
    }
    else if(age < age2){
        std::cout << name << " is younger than " << name2 << std::endl;
    }
    else{
        std::cout << name << " is the same age as " << name2 << std::endl;
    }
    return 0;
}

File: Projects/01-Name_And_Age/Rust/main.rs
use std::io;
use std::io::Write;
use std::num::ParseIntError;

fn user_input_str() -> Result<String, io::Error> {
    let mut buff = String::new();
    let _rl_result = io::stdin().read_line(&mut buff);
    Ok(buff
        .trim()
        .to_string())
}

fn user_input_i32() -> Result<i32, ParseIntError> {
    let mut buff = String::new();
    let _rl_result = io::stdin().read_line(&mut buff);
    Ok(buff.trim()
           .parse()?)
}

fn flush() {
    io::stdout().flush()
                .expect("stdout ran into an error while flushing.")
}

fn main() {
    print!("Please enter your name: ");
    flush();
    let name = user_input_str().expect("String could not be acquired from stdin.");

    print!("Please enter your age: ");
    flush();
    let age = user_input_i32().expect("Integer could not be acquired from stdin.");

    let name2 = "Peter Parker";
    let age2 = 17;

    if age > age2 {
        println!("{} is older than {}.", name, name2);
    }
    else if age < age2 {
        println!("{} is younger than {}.", name, name2);
    }
    else {
        println!("{} is the same age as {}.", name, name2);
    }
}
